### PR TITLE
DDCNL-4606: remove unwanted exceptions

### PR DESCRIPTION
**Pull request by lewey**

### CODE REVIEW

File: app/uk/gov/hmrc/nationalinsurancerecord/connectors/DesConnector.scala [hunk 1]
```diff
@@ -120,18 +120,16 @@ class DesConnector @Inject()(
...
+        Future.successful(Right(responseModel))
       case None =>
-        connectToDes(url, api)(hc, formatA).flatMap {
-          case Right(response) =>
-            logger.info(s"*~* - writing nino to cache: $nino")
-            repository.insertByNino(nino, response)
-            Future.successful(response)
-          case Left(error) =>
-            Future.failed(error)
+         connectToDes(url, api)(hc, formatA) map { _ map { validResponse =>
```
Comment (vinnicombe) [inline app/uk/gov/hmrc/nationalinsurancerecord/connectors/DesConnector.scala:128]: Why is map { _ map { used over flatMap?

File: app/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordService.scala [hunk 2]
```diff
@@ -39,32 +39,43 @@ class NationalInsuranceRecordService @Inject()(
...
     featureFlagService.get(ProxyCacheToggle) flatMap {
       proxyCache =>
         citizenDetailsService.checkManualCorrespondenceIndicator(nino) flatMap {
           mci =>
             if (proxyCache.isEnabled) {
-              proxyCacheConnector.get(nino) map {
-                pcd =>
-                  buildNationalInsuranceRecord(pcd.niRecord, pcd.summary, pcd.liabilities, mci)
-              }
+              proxyCacheConnector.get(nino) map (_.map (pcd => buildNationalInsuranceRecord(pcd.niRecord, pcd.summary, pcd.liabilities, mci)))
```
Comment (vinnicombe) [inline app/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordService.scala:48]: Similar question here, but guessing this returns an Either and is right biased?

File: app/uk/gov/hmrc/nationalinsurancerecord/connectors/DesConnector.scala (hunk 1 above)
Comment (vinnicombe) [reply app/uk/gov/hmrc/nationalinsurancerecord/connectors/DesConnector.scala:128]: Because it's an Either and right biased

**Review by vinnicombe — APPROVED** [review]

File: app/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordService.scala (hunk 2 above)
Comment (lewey) [reply app/uk/gov/hmrc/nationalinsurancerecord/services/NationalInsuranceRecordService.scala:48]: Yes

**Pull request merged**